### ISSUE TITLE
[QUESTION] gradient and control.joint_act

### ISSUE THREAD

**Issue by ucacaxm** [NVIDIA/warp#443]
Hi all,

I try to optimize the control of a character with `joint_act` but the gradient is always 0.
To test the problem, I have only three rigid bodies and 2 joints.

I use `control.joint_act`. 
```
        self.model = builder.finalize(requires_grad=True)
        self.control = self.model.control() 
        self.control.joint_act.requires_grad = True
```
The `joint_axis_mode`is set to `JOINT_MODE_TARGET_POSITION` for all joints.

The action values has the `required_grad` to True.
```
        self.actions = wp.array( init_action, dtype=wp.float32, requires_grad=True )     # init_action is a numpy array
```

The loss computation is simple (just want test). 
But the actions gradient is always 0 (I print them) 
```
   print("action grad", self.actions.grad.numpy())
   print("control.joint_act grad:", self.control.joint_act.grad.numpy())
```
The gradient of the `body_q` are ok (not equal to 0 and coherent), so a gradient is computed but it seems it does not backward until the action values.

Is there something special with the gradient and joint_act (control) ? (Or I have a bug).

Many thanks ...

**Comment by ucacaxm** [comment]
I found it.
I got my own control during the creation
```
   self.control = self.model.control() 
```

I have to pass it as argument of `simulate`: 
```
   self.integrator.simulate(self.model, self.states[i], self.states[i+1], self.sim_dt**, self.control**)
```
I thought that the default arg would be the same `control` I got during the creation.

Best,
